clearwater-sip-stress.root/usr/share/clearwater/bin/run_stress

```diff
@@ -0,0 +1,151 @@
+parser.add_argument('subscribers', help='Number of subscribers to emulate', type=int)
+parser.add_argument('duration', help='Number of minutes to run stress for', type=int)
+parser.add_argument('--multiplier', help='Multiplier for the VoLTE load profile (e.g. passing 2 here will mean 2.6 calls and 4 re-registers per sub per hour)', type=int, default=1)
+parser.add_argument('--initial-reg-rate', help='Rate to send in initial, pre-test-run registrations', type=int, default=80)
+parser.add_argument('--target', help='Domain/IP and port to target stress at. Default is sprout.{domain}:5054')
+
+args = parser.parse_args()
+
+if args.target is None:
+    args.target = "sprout.{}:5054".format(args.domain)
+
+# Calculate calls/registers per second based on the input
+CPS = BHCA * args.subscribers * args.multiplier / 3600
+RPS = REREG_PER_HOUR * args.subscribers * args.multiplier / 3600
+
+# Given the duration, how many registers/calls do we need to make?
+REG_MAX = int(args.duration * 60 * RPS)
+CALL_MAX = int(args.duration * 60 * CPS)
+
+SUB_START = 2010000000
+SUB_END = SUB_START + args.subscribers
+
+registers = range(SUB_START, SUB_END)
+callers = range(SUB_START, SUB_END, 2)
+
+# Randomly shuffle the order of registers and calls to reduce the risk of any cache issues
+shuffle(registers)
+shuffle(callers)
+
+# Generate the CSV input files - one for REGISTER messages with one user per
+# line, one for INVITE flows with a caller/callee pair on each line
+with open(REGISTER_CSV, "w") as f:
+    f.write("SEQUENTIAL\n")
+    for s in registers:
+        f.write("{}\n".format(s))
+
+with open(INVITE_CSV, "w") as f:
+    f.write("SEQUENTIAL\n")
+    for s in callers:
+        f.write("{};{}\n".format(s, s+1))
+
+null = open("/dev/null", "w+")
+
+common_args = ["-key", "home_domain", args.domain,
+               "-t", "t1",
+               args.target]
+
+initial_reg_cmd = [SIPP_BIN,
+                   "-sf", SCRIPTS_DIR + "register.xml",
+                   "-inf", REGISTER_CSV,
+                   "-r", str(args.initial_reg_rate),
+                   "-m", str(args.subscribers)] + common_args
+
+rereg_cmd = [SIPP_BIN,
+             "-sf", SCRIPTS_DIR + "register.xml",
+             "-inf", REGISTER_CSV,
+             "-r", str(RPS),
+             "-m", str(REG_MAX)] + common_args
+
+caller_cmd = [SIPP_BIN,
+              "-sf", SCRIPTS_DIR + "caller.xml",
+              "-inf", INVITE_CSV,
+              "-r", str(CPS),
+              "-d", "5000",
+              "-m", str(CALL_MAX)] + common_args
+
+# Callee must listen on port 5082, because this is what's in the Path header in
+# register.xml.
+
+callee_cmd = [SIPP_BIN,
+              "-sf", SCRIPTS_DIR + "callee.xml",
+              "-p", "5082",
+              "-t", "t1",
+              "-m", str(CALL_MAX)]
+
+# Do initial registrations as quickly as possible for test setup - this is an
+# unrealistic load profile, so no need for measurements yet.
```

Comment:
What if some of these initial registers fail though? That will inevitably cause future failures, right? It would be nice to easily identify this. This will possibly be covered in your future output work.